pkg/resource/notebook_instance/testdata/test_suite.yaml

```diff
@@ -0,0 +1,358 @@
+       given:
+         desired_state: "notebook_instance/v1alpha1/input/nb_invalid_before_create.yaml"
+         svc_api:
+           - operation: CreateNotebookInstanceWithContext
+             error:
+               code: InvalidParameterValue
+               message: "The notebook instance name must start with an alphanumeric character."
+       invoke: Create
+       expect:
+         latest_state: "notebook_instance/v1alpha1/output/nb_invalid_after_create.yaml"
+         error: resource is in terminal condition
+     - name: "Create=Valid"
+       description: "Testing a successful create call"
+       given:
+         desired_state: "notebook_instance/v1alpha1/input/nb_success_before_create.yaml"
+         svc_api:
+           - operation: CreateNotebookInstanceWithContext
+             output_fixture: "notebook_instance/create/create_success.json"
+       invoke: Create
+       expect:
+         latest_state: "notebook_instance/v1alpha1/output/nb_success_after_create.yaml"
+  - name: "Notebook Instance readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+       - name: "ReadOne=AfterCreate"
+         description: "Testing readOne right after create, the status should be in Pending with Resource synced being false."
+         given: 
+           desired_state: "notebook_instance/v1alpha1/input/readone/nb_readone_pending_after_create.yaml"
+           svc_api:
+             - operation: DescribeNotebookInstanceWithContext
+               output_fixture: "notebook_instance/readOne/readone_pending_after_create.json"
+         invoke: ReadOne
+         expect:
+           latest_state: "notebook_instance/v1alpha1/output/readone/nb_readone_pending_after_create.yaml"
+       - name: "ReadOne=Deleting"
+         description: "Testing readOne when deleting, resource synced should be false."
+         given: 
+           desired_state: "notebook_instance/v1alpha1/input/readone/nb_readone_deleting.yaml"
+           svc_api:
+             - operation: DescribeNotebookInstanceWithContext
+               output_fixture: "notebook_instance/readOne/readone_deleting.json"
+         invoke: ReadOne
+         expect:
+           latest_state: "notebook_instance/v1alpha1/output/readone/nb_readone_deleting.yaml"
+       - name: "ReadOne=Fail"
+         description: "Testing readOne when Describe fails."
+         given: 
+           desired_state: "notebook_instance/v1alpha1/input/readone/nb_readone_fail.yaml"
+           svc_api:
+             - operation: DescribeNotebookInstanceWithContext
+               error:
+                code: ServiceUnavailable
+                message: "Server is down"
+         invoke: ReadOne
+         expect:
+           error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+       - name: "ReadOne=FailedState"
+         description: "Testing how readone handles the Failed State, it should have a failure reason."
```

Comment:
nit: inconsistent naming of `readone`. In most other descriptions you use `readOne`.

Reply:
Going to use readone from now on just to be consistent.